completor/main.py

```diff
@@ -253,12 +249,11 @@ def create(
             else:
                 progress_bar.update(len(lines) - prev_line_number)
 
-        for well_name_ in well_names:
+        for i, well_name_ in enumerate(well_names):
             logger.debug("Writing new MSW info for well %s", well_name_)
-            wells = Wells(well_name_, case, schedule_data)
-            well_number = read_schedule.get_well_number(well_name_, active_wells)
-            output = CreateOutput(case, schedule_data, wells, well_name_, well_number, show_fig, figure_name, paths)
-            output_text += format_text(None, output.finalprint)
+            well = Well(well_name_, i, case, schedule_data[well_name_])
+            output = create_output.format_output(well, figure_name, paths)
+            output_text += format_text(None, output)
 
     except Exception as e_:
```

Comment:
Think this can be replaced by
````
    except Exception as e_:
        # Make sure the output thus far is written, and figure files are closed.
        output_text = _replace_preprocessing_names(output_text, case.mapper)
        with open(new_file, "w", encoding="utf-8") as file:
            file.write(output_text)
        raise e_
        

````